spec/model/github_installation_spec.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+require_relative "spec_helper"
+
+RSpec.describe GithubInstallation do
+  subject(:installation) {
+    project = Project.create_with_id(name: "default", provider: "hetzner").tap { _1.associate_with_project(_1) }
+
+    described_class.create_with_id(installation_id: 123, project_id: project.id, name: "test-user", type: "User")
+  }
+
+  it "returns sum of used vm cores" do
+    vm1 = Vm.create_with_id(display_state: "running", unix_user: "x", public_key: "x", name: "x", family: "standard", cores: 2, location: "x", boot_image: "x")
+    vm2 = Vm.create_with_id(display_state: "running", unix_user: "x", public_key: "x", name: "x", family: "standard", cores: 4, location: "x", boot_image: "x")
+    Vm.create_with_id(display_state: "running", unix_user: "x", public_key: "x", name: "x", family: "standard", cores: 8, location: "x", boot_image: "x")
```

Comment:
I might try to factor out the dynamic part of this, because it's an eye maze to figure out the variant part.